internal/sliceutils/sliceutils.go

```diff
@@ -77,3 +77,73 @@ func AppendWithoutDupes(a []string, b ...string) []string {
 
 	return ret
 }
+
+type EscapeSplitOptions struct {
+	Delimiter  rune
+	EscapeChar rune
+}
+
+func NewEscapeSplitOptions() EscapeSplitOptions {
+	return EscapeSplitOptions{
+		Delimiter:  ':',
+		EscapeChar: '\\',
+	}
+}
+
+func UnescapeAndSplit(s string, opts EscapeSplitOptions) ([]string, error) {
+	var parts []string
+	escapeNextChar := false
+	cur := ""
+	for i, r := range s {
+		if escapeNextChar {
+			if r == opts.Delimiter || r == opts.EscapeChar {
+				cur += string(r)
+			} else {
+				// grab the 6 chars around the invalid char to make it easier to find the offending string from the error
+				snippetStart := i - 3
+				if snippetStart < 0 {
+					snippetStart = 0
+				}
+				snippetEnd := i + 3
+				if snippetEnd > len(s) {
+					snippetEnd = len(s)
+				}
+
+				return nil, fmt.Errorf("invalid escape sequence '%c%c' in '%s'", opts.EscapeChar, r, s[snippetStart:snippetEnd])
```

Comment:
question: this syntax already allows regexps between the colons right? doesn't that mean you can't escape things for the regexp anymore?

Reply:
Nope - the `ObjectSelector` type allows regexes, but the function that converts this syntax into an `ObjectSelector` [calls `regexp.QuoteMeta`](https://github.com/tilt-dev/tilt/blob/816e9a6a9d150f8b543c7bb2112a2c558caff0cd/internal/k8s/object_selector.go#L100) on all of its components.

Other places that use the `ObjectSelector` type and do allow regexes (also highlighting the need for cleaning this up): `k8s_kind`, `k8s_image_json_path`, `filter_yaml`